lib/internal/url.js

```diff
@@ -1367,13 +1367,18 @@ function getPathFromURLPosix(url) {
   return decodeURIComponent(pathname);
 }
 
-function getPathFromURL(path) {
+function getPathFromURL(path, strictPath) {
   if (path == null || !path[searchParams] ||
       !path[searchParams][searchParams]) {
     return path;
   }
   if (path.protocol !== 'file:')
     return new errors.TypeError('ERR_INVALID_URL_SCHEME', 'file');
+  if (strictPath && typeof path.href === 'string' &&
```

Comment:
The `typeof path.href === 'string'` should be unnecessary because a valid WHATWG URL should always have a string as `href`